From ffcc8a9e05caec1def8db8f5218e2ecad264a2d4 Mon Sep 17 00:00:00 2001
From: Lukas Tribus <luky-37@hotmail.com>
Date: Tue, 26 Jan 2016 20:33:14 +0100
Subject: [PATCH 29/67] MINOR: unix: don't mention free ports on EAGAIN

When a connect() to a unix socket returns EAGAIN we talk about
"no free ports" in the error/debug message, which only makes
sense when using TCP.

Explain connect() failure and suggest troubleshooting server
backlog size.
(cherry picked from commit 9f256d4d85f2b9c79b0d4f28a27731260eefa1b2)
---
 src/proto_uxst.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/proto_uxst.c b/src/proto_uxst.c
index 7cc63c0..b2a7fe2 100644
--- a/src/proto_uxst.c
+++ b/src/proto_uxst.c
@@ -504,7 +504,7 @@ int uxst_connect_server(struct connection *conn, int data, int delack)
 		else if (errno == EAGAIN || errno == EADDRINUSE || errno == EADDRNOTAVAIL) {
 			char *msg;
 			if (errno == EAGAIN || errno == EADDRNOTAVAIL) {
-				msg = "no free ports";
+				msg = "can't connect to destination unix socket, check backlog size on the server";
 				conn->err_code = CO_ER_FREE_PORTS;
 			}
 			else {
-- 
2.4.10

